kernel/kernel-api/src/test/scala/io/delta/kernel/internal/replay/ActionsIteratorSuite.scala

```diff
@@ -0,0 +1,106 @@
+
+  /**
+   * Test for ActionsIterator resource leak fix validation
+   *
+   * This test validates that the fix applied in ActionsIterator.java prevents resource
+   * leaks by ensuring that CloseableIterators are properly closed when exceptions occur.
+   *
+   * The specific fix being tested: Utils.closeCloseablesSilently(dataIter) in the catch block of
+   * readCommitOrCompactionFile method.
+   */
+  test("ActionsIterator readCommitOrCompactionFile resource cleanup") {
+    var iteratorClosed = false
+
+    class TestEngine extends Engine {
+      override def getExpressionHandler: ExpressionHandler =
+        throw new UnsupportedOperationException("Not needed for this test")
+      override def getParquetHandler: ParquetHandler =
+        throw new UnsupportedOperationException("Not needed for this test")
+      override def getFileSystemClient: FileSystemClient =
+        throw new UnsupportedOperationException("Not needed for this test")
+      override def getJsonHandler: JsonHandler = new JsonHandler {
+        override def readJsonFiles(
+            fileIter: CloseableIterator[FileStatus],
+            physicalSchema: StructType,
+            predicate: Optional[Predicate]): CloseableIterator[ColumnarBatch] = {
+          // Return an empty iterator that tracks closure
+          new CloseableIterator[ColumnarBatch] {
+            override def hasNext(): Boolean =
+              throw new NoSuchElementException("This is a test exception")
+            override def next(): ColumnarBatch =
+              throw new UnsupportedOperationException("Not needed for this test")
+            override def close(): Unit = iteratorClosed = true
+          }
+        }
+        override def parseJson(
+            jsonStringVector: ColumnVector,
+            outputSchema: StructType,
+            selectionVector: Optional[ColumnVector]): ColumnarBatch = {
+          throw new UnsupportedOperationException("Not needed for this test")
+        }
+        override def writeJsonFileAtomically(
+            filePath: String,
+            dataIter: CloseableIterator[Row],
+            overwrite: Boolean): Unit = {
+          throw new UnsupportedOperationException("Not needed for this test")
+        }
+      }
+    }
+    val engine = new TestEngine()
+
+    val testFile = FileStatus.of(
+      "/path/to/00000000000000000000.json",
+      100L,
+      System.currentTimeMillis())
+    val files = Collections.singletonList(testFile)
+    val schema = new StructType()
+
+    val actionsIterator =
+      new ActionsIterator(engine, files, schema, Optional.empty[Predicate]())
+
+    try {
+      actionsIterator.hasNext()
+      fail("Expected NoSuchElementException was not thrown")
+    } catch {
+      case _: NoSuchElementException =>
+    }
+
+    // Closing the ActionsIterator does not close the internal iterator
+    // Because of the exception the internal iterator is lost
+    actionsIterator.close()
```

Comment:
put it after the final assertion or remove. It brings more confusion to the reader

Reply:
The intention here is showing that even after actionIterator.close is called, the iteratorClosed that tracks the internal iterator is false.

If it is confusing I can remove it. Let me know

Reply:
Done